modules/navier_stokes/include/fvkernels/PINSFVMomentumPressureFlux.h

```diff
@@ -10,17 +10,22 @@
 #pragma once
 
 #include "FVFluxKernel.h"
+#include "INSFVMomentumResidualObject.h"
 
 /**
  * A flux kernel using the divergence theorem for the pressure gradient term in the momentum
  * equation
  */
-class PINSFVMomentumPressureFlux : public FVFluxKernel
+class PINSFVMomentumPressureFlux : public FVFluxKernel, public INSFVMomentumResidualObject
 {
 public:
   static InputParameters validParams();
   PINSFVMomentumPressureFlux(const InputParameters & params);
 
+  // Pressure term so no RC data involved
+  void gatherRCData(const Elem &) override {}
+  void gatherRCData(const FaceInfo &) override {}
```

Comment:
```suggestion
  void gatherRCData(const Elem &) override final {}
  void gatherRCData(const FaceInfo &) override final {}
```

Reply:
I guess we could just make that inherit from INSFVMomentumPressureFlux at this point

Reply:
that object doesn't exist